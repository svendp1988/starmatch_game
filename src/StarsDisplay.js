import React from "react";
import {utils} from "./utils";

export class StarsDisplay extends React.Component {
    render() {
        return (
            <>
                {utils.range(1, this.props.count)
                    .map(starId => <div key={starId} className="star"/>)}
            </>
        )
    }
}


