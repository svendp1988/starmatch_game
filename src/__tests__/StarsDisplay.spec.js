import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {StarsDisplay} from "../StarsDisplay";

describe('StarsDisplay', function () {
    const props = {
        count: 9
    }

    it('should match snapshot', function () {
        const tree = renderer.create(
            <StarsDisplay {...props}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should contain 9 divs with a class of star', function () {
        const wrapper = shallow(<StarsDisplay {...props} />);
        expect(wrapper.children()).toHaveLength(9);
        wrapper.children()
            .forEach(child =>
                expect(child.props().className).toEqual("star"));
    });
});
