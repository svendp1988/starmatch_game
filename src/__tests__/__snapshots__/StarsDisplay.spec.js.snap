// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarsDisplay should match snapshot 1`] = `
Array [
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
  <div
    className="star"
  />,
]
`;
